src/EFCore.PG/Extensions/NpgsqlMetadataExtensions.cs

```diff
@@ -44,5 +45,11 @@ public static INpgsqlPropertyAnnotations Npgsql([NotNull] this IProperty propert
 
         public static NpgsqlPropertyAnnotations Npgsql([NotNull] this IMutableProperty property)
             => (NpgsqlPropertyAnnotations)Npgsql((IProperty)property);
+
+        public static INpgsqlRelationalAnnotations Npgsql([NotNull] this IAnnotatable annotatable)
```

Comment:
See note below on annotations belonging in NpgsqlModelAnnotations.

These lines are definitely not a good idea, they allow you to easily add annotations meant on the model on any other type (e.g. define a PostgreSQL extension on a property).

Reply:
The issue is that the annotations of interest are passed from the model on to derivatives of `MigrationOperation` which is an `Annotatable`, and that's where we were lacking idiomatic accessors.

I could rework these to live in something like `IMigrationOperationAnnotations`, but there's still the question of how to handle the `AlterDatabaseOperation.OldDatabase` property which is a plain `Annotatable`. So that's why I was trying to set up an accessor for basically any annotation.

As a first suggestion, I could remove the mutable version here, so that this only returns a read-only version... If that still sounds too risky/improper, I can take another look at the big picture to see if this could be accomplished with `MigrationOperation`-based annotations